src/attention/utils/helpers.ts

```diff
@@ -1,98 +1,240 @@
-  arrowEl?: HTMLElement | null;
-  attentionEl?: HTMLElement | null;
-  targetEl?: unknown;
-  top?: Boolean;
-  right?: Boolean;
-  bottom?: Boolean;
-  left?: Boolean;
-  tooltip?: Boolean;
-  popover?: Boolean;
-  callout?: Boolean;
-  noArrow?: Boolean;
-  waitForDOM?: () => void;
-};
-
-const middlePosition = "calc(50% - 7px)";
-const isDirectionVertical = (name: string) => [TOP, BOTTOM].includes(name);
-function computeCalloutArrow({
-  actualDirection,
-  directionName,
-  arrowEl,
-}: AttentionState) {
-  if (!arrowEl) return;
-
-  actualDirection = directionName;
-  const directionIsVertical = isDirectionVertical(directionName);
-  arrowEl.style.left = directionIsVertical ? middlePosition : "";
-  arrowEl.style.top = !directionIsVertical ? middlePosition : "";
+  isShowing?: boolean
+  isCallout?: boolean
+  actualDirection?: Directions
+  directionName?: Directions
+  arrowEl?: HTMLElement | null
+  attentionEl?: HTMLElement | null
+  flip?: Boolean
+  fallbackPlacements?: Directions[]
+  targetEl?: ReferenceElement | null
+  noArrow?: Boolean
+  distance?: number
+  skidding?: number
+  waitForDOM?: () => void
 }
 
-export async function useRecompute(state: AttentionState) {
-  if (!state.isShowing) return; // we're not currently showing the element, no reason to recompute
-  await state?.waitForDOM?.(); // wait for DOM to settle before computing
-  if (state.isCallout) return computeCalloutArrow(state); // we don't move the callout box, only its arrow
-  const position = await computePosition(
-    state.targetEl as ReferenceElement,
-    state.attentionEl as HTMLElement,
-    {
-      placement: state.directionName,
-      middleware: [
-        // Should we make this configurable, but have these as sane defaults?
-        flip(),
-        offset(8),
-        shift({ padding: 16 }),
-        // @ts-ignore
-        arrow({ element: state.noArrow ? undefined : state.arrowEl }), // FIXME
-      ],
+const middlePosition: string = 'calc(50% - 7px)'
+const isDirectionVertical = (name: Directions): boolean =>
+  ([TOP_START, TOP, TOP_END, BOTTOM_START, BOTTOM, BOTTOM_END] as Directions[]).includes(name)
+
+export const arrowDirectionClassname = (dir: Directions) => {
+    let direction: Directions
+    if (/-/.test(dir)) {
+      direction = dir
+        .split('-')
+        .map((d) => d.charAt(0).toUpperCase() + d.slice(1))
+        .join('') as Directions
+    } else {
+      direction = dir.charAt(0).toUpperCase() + dir.slice(1) as Directions
     }
-  );
-  // @ts-ignore
-  state.actualDirection = position.placement;
-  Object.assign(state.attentionEl?.style || {}, {
-    left: "0",
-    top: "0",
-    transform: `translate3d(${Math.round(position.x)}px, ${Math.round(
-      position.y
-    )}px, 0)`,
-  });
-  // @ts-ignore
-  let { x, y } = position.middlewareData.arrow;
-
-  if (state.arrowEl) {
-    state.arrowEl.style.left = x ? x + "px" : "";
-    state.arrowEl.style.top = y ? y + "px" : "";
+    return `${direction}`
```

Comment:
Can it just be `return direction`?

Reply:
You're right! 👍 I must have forgotten to change it when I refactored it before! 🙈

Reply:
Updated in this commit: [5440729](https://github.com/warp-ds/core/pull/15/commits/54407296b1cb66809b04d2f2312b47026eff4ff8)